### PR TITLE
Schemas: add `/schemas/list` page, and custom 404 for `/schemas` and missing subpages

### LINKED ISSUE #1102: Right nav formatting problem when page has no TOC
E.g., see:

- https://opentelemetry.io/docs/getting-started/
- https://opentelemetry.io/blog/2023/integrations/
- https://opentelemetry.io/docs/instrumentation/ruby/events/

Here's an example:

> Screen Shot 2022-02-04 at 9 37 27 AM

Notice the space missing between the page-meta links and the top-nav.

Might this be a recently introduced docsy issue?

Here is the corresponding Docsy issue:

- https://github.com/google/docsy/issues/1573

### LINKED ISSUE #3468: `/schemas` is the top 404
Clearly folks are interested in listing the available schemas, because `/schemas` (or `/schemas/`) is the #1 legitimate 404. I think that there should be a schemas index page, and will submit a proposal soon.

@tigrannajaryan had already argued against an index page, see https://github.com/open-telemetry/opentelemetry.io/issues/1144#issuecomment-1041845740. I'd like to open this up for discussion and see if we can find a win-win solution.

Related:

- #3256

/cc @open-telemetry/specs-semconv-maintainers @jsuereth @svrnm @MrAlias

### PR DESCRIPTION
**Pull request by chalin**
@svrnm @tigrannajaryan I think that this is most likely the solution that could meet all of [our concerns](https://github.com/open-telemetry/opentelemetry.io/issues/3468#issuecomment-1798307259), as summarized by @svrnm. Here is what this PR offers:

- Fixes #3468
- Adds a `/schemas/list` page that lists the published schemas.
- Adds a custom 404 page for `/schemas` that contains the link to the list page
- Fixes #1102

**Previews**

- https://deploy-preview-3508--opentelemetry.netlify.app/schemas - shows custom schemas 404
- https://deploy-preview-3508--opentelemetry.netlify.app/schemas/ - shows custom schemas 404
- https://deploy-preview-3508--opentelemetry.netlify.app/schemas/list/ - shows schemas list page

**Redirect tests**

- https://deploy-preview-3508--opentelemetry.netlify.app/schemas/latest --> 1.22.0 (as of the time of writing)
- https://deploy-preview-3508--opentelemetry.netlify.app/schemas/latest/ --> 1.22.0 (as of the time of writing)
- https://deploy-preview-3508--opentelemetry.netlify.app/schemas/x.y.z - nonexistent version --> schemas 404

**Status tests**

```console
$ curl -sI https://deploy-preview-3508--opentelemetry.netlify.app/schemas | grep -E 'HTTP|-type|location' 
HTTP/2 404 
content-type: text/html; charset=UTF-8
$ curl -sI https://deploy-preview-3508--opentelemetry.netlify.app/schemas/ | grep -E 'HTTP|-type|location'
HTTP/2 404 
content-type: text/html; charset=UTF-8
$ curl -sI https://deploy-preview-3508--opentelemetry.netlify.app/schemas/list/ | grep -E 'HTTP|-type|location' 
HTTP/2 200 
content-type: text/html; charset=UTF-8
```

### Screenshots

404 page:

> 
image

List page:

> image

### CODE REVIEW

File: content/en/schemas/list.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+linkTitle: List
+type: docs
+body_class: ot-schema-list
+---
+
+The OpenTelemetry schemas hosted on this site are listed below. To learn more
+about schemas, see [Telemetry Schemas][].
+
+{{% schemas %}}
+
```
Comment (svrnm) [inline content/en/schemas/list.md:12]:
```suggestion

> **Note**: This page is not intended to be consumed by automatic processes, as the format may change.
```

I think also this page should have such a disclaimer that this page is not here for being consumed by any kind of automatism.

**Review by svrnm — CHANGES_REQUESTED** [review]
lgtm, some minor suggestions

File: content/en/schemas/list.md (hunk 1 above)
Comment (chalin) [reply content/en/schemas/list.md:12]: This list page is a normal web being served by the site. All pages are subject to change :). I'd vote against such a note, but I'll add one anyways and you can let me know what you think.

**Comment by chalin** [comment]
Closing in favor of:

- #3537

**Pull request closed without merging**